lms/static/scripts/frontend_apps/components/dashboard/DashboardActivityFilters.tsx

```diff
@@ -12,6 +16,7 @@ export type DashboardActivityFiltersProps = {
   onAssignmentsChange: (newAssignmentIds: string[]) => void;
   selectedStudentIds: string[];
   onStudentsChange: (newStudentIds: string[]) => void;
+  onSelectionReset?: () => void;
```

Comment:
At first I was planning to simply do something in these lines:

```ts
onClick={() => {
  onCoursesChange([]);
  onAssignmentsChange([]);
  onStudentsChange([]);
}}
```

However, since all these end up calling `useDashboardFilters`' `updateFilters`, which does not use regular state that can be "queued" by preact, only the last one ended up being applied.

Instead, I added an optional extra callback, so that parent components can decide what to do to clear filters, which in the case of persisting in the query string is `updateFilters({ studentIds: [], assignmentIds: [], courseIds: [] })`, in a single operation.

Reply:
> However, since all these end up calling useDashboardFilters' updateFilters, which does not use regular state that can be "queued" by preact, only the last one ended up being applied.

Hmm... this seems like a hazard that could trip up some other code in future. From a glance at the implementation my understanding is that the issue is that `updateFilters` relies on the `queryParams` value from the last render, instead of always reading the latest params. One way to avoid that would be to always read the query params directly from the URL.

Having a single callback that can reset everything in one go still seems desirable from an efficiency point of view.